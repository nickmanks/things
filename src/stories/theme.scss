
$storybook-container-shadow: 1px 1px 5px 1px rgba(0, 0, 0, .37);

.story-container {
  margin: 3rem;
  box-shadow: $storybook-container-shadow;
  width: 100%;
  height: 65rem;
  border-radius: .4rem;

  > div {
    height: 100%;
    width: 100%;

    > div:first-child {
      height: 100%;
      width: 100%;
    }
  }
}
