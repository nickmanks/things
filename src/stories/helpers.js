import React from 'react';
import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import reducers from '../reducers';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'shards-ui/dist/css/shards.min.css';
import './theme.scss';


export function withCustomContainer(backgroundColor) {
  const CustomContainer = (story)=> (
    <div className={'story-container'} style={{backgroundColor}}>
      {story()}
    </div>
  );
  CustomContainer.displayName = 'CustomContainer';
  return CustomContainer;
}

export const storyStore =
  (state)=> createStore(reducers, state, applyMiddleware(thunk));
