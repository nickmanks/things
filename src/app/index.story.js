/* global module */
import React from 'react';
import {Provider} from 'react-redux';
import {storiesOf} from '@storybook/react';
import {storyStore} from '../stories/helpers';
import App from '.';
import testItems from '../testing/test-items';


storiesOf('Connected/App', module)
  .add('with multiple items', ()=> {
    const store = storyStore({
      things: {items: testItems},
      persistence: {
        loaded: true,
        queued: [],
        processing: [],
        done: []
      }
    });

    return (
      <Provider store={store}>
        <App store={store} />
      </Provider>
    );
  })
  .add('persistence loading', ()=> {
    const store = storyStore({
      things: {items: testItems}
    });

    return (
      <Provider store={store}>
        <App store={store} />
      </Provider>
    );
  });
