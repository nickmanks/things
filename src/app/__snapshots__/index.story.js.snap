// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Connected/App with multiple items 1`] = `
<App store={{}} loaded={false} dispatch={"function"} storeSubscription={{"store":{},"listeners":{}}}>
  <Fragment>
    <Connect(AppBar) />
    <div className={"app-loading-spinner"}>
      <div className={"app-spinner"} />
    </div>
    <Connect(PersistenceManager) />
  </Fragment>
</App>
`;
